deployment/keystone/deploy.go

```diff
@@ -137,6 +145,85 @@ func DeployContracts(lggr logger.Logger, e *deployment.Environment, chainSel uin
 	}, nil
 }
 
+// DonInfo is DonCapabilities, but expanded to contain node information
+type DonInfo struct {
+	Name         string
+	Nodes        []Node
+	Capabilities []kcr.CapabilitiesRegistryCapability // every capability is hosted on each node
+}
+
+type Node struct {
+	ID           string
+	P2PID        string
+	Name         string
+	PublicKey    *string
+	ChainConfigs []*nodev1.ChainConfig
+}
+
+func NodesFromJD(name string, nodeIDs []string, jd deployment.OffchainClient) ([]Node, error) {
```

Comment:
ditto here should be a way to either combine with or have another helper alongside https://github.com/smartcontractkit/chainlink/blob/c6c8fbd559091bd3fb6673c7fb20c90a92aebe1d/deployment/environment.go#L247

Reply:
The code in there was very CCIP specific, only supporting EVM which needs a larger cleanup. Keystone also uses peer IDs for node IDs so our lookup differs. I'd prefer to leave a TODO here and fix this in a follow up

Reply:
I can see it only support EVM at the moment, but what part is CCIP specific?